### PR TITLE
feature/LeaveApplication

### PR DESCRIPTION
**Pull request by FredMadrona**
🎫 Ticket

[Leave Application ](https://trello.com/c/Fn8bkj64/72-filament-leave-application)

📦 Packages

none

📝 Updates

CRUD functionality for Leave Application section admin panel

🧪 Testing Steps

login to admin panel and select Leave Application section
image

### CODE REVIEW

File: src/app/Models/LeaveApplication.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    use HasFactory;
+
+    protected $table = 'leave_applications';
+
+    protected $fillable = [
+        'employee_name',
+        'start_date',
+        'end_date',
+        'leave_type',
```
Comment (ejempic) [inline src/app/Models/LeaveApplication.php:18]: leave type should be enum

File: src/app/Models/LeaveApplication.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        'reason',
+        'status',
```
Comment (ejempic) [inline src/app/Models/LeaveApplication.php:20]: general enum type int

File: src/database/seeders/LeaveApplicationSeeder.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public function run()
+    {
+        LeaveApplication::create([
+            'employee_name' => 'John Doe',
+            'start_date' => Carbon::now()->addDays(10),
+            'end_date' => Carbon::now()->addDays(15),
+            'leave_type' => 'Vacation',
+            'reason' => 'Family vacation',
+            'status' => 'Pending',
```
Comment (ejempic) [inline src/database/seeders/LeaveApplicationSeeder.php:19]: pag enum na 
```suggestion
            'status' => STATUS::PENDING,
```

File: src/database/seeders/LeaveApplicationSeeder.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class LeaveApplicationSeeder extends Seeder
+{
+    public function run()
+    {
+        LeaveApplication::create([
+            'employee_name' => 'John Doe',
+            'start_date' => Carbon::now()->addDays(10),
+            'end_date' => Carbon::now()->addDays(15),
+            'leave_type' => 'Vacation',
```
Comment (ejempic) [inline src/database/seeders/LeaveApplicationSeeder.php:17]:
```suggestion
            'leave_type' => LEAVE_TYPE::VACATION,
```

**Review by ejempic — CHANGES_REQUESTED** [review]

File: src/app/Enums/LeaveStatus.php [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Enums;
+
+enum LeaveStatus: int
+{
+    case Pending = 1;
+    case Approved = 2;
+    case Rejected = 3;
+}
```
Comment (Kelv1nG) [inline src/app/Enums/LeaveStatus.php:10]: enum value should still be a string,

case PENDING = 'pending' etc

avoid using magic numbers

File: src/app/Enums/LeaveType.php [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace App\Enums;
+
+enum LeaveType: int
+{
+    case SICK_LEAVE = 1;
+    case VACATION_LEAVE = 2;
+    case ANNUAL_LEAVE = 3;
+    case MATERNITY_LEAVE = 4;
+}
```
Comment (Kelv1nG) [inline src/app/Enums/LeaveType.php:11]: enum value should still be a string,

case SICK_LEAVE = 'sick_leave' etc

avoid using magic numbers

File: src/database/migrations/2024_11_12_131427_create_leave_applications_table.php [hunk 7]
```diff
@@ -12,9 +12,9 @@ public function up(): void
             $table->string('employee_name');
             $table->date('start_date');
             $table->date('end_date');
-            $table->string('leave_type');
+            $table->unsignedSmallInteger('leave_type');
```
Comment (Kelv1nG) [inline src/database/migrations/2024_11_12_131427_create_leave_applications_table.php:15]: can set as enum

File: src/database/migrations/2024_11_12_131427_create_leave_applications_table.php [hunk 8]
```diff
@@ -12,11 +12,12 @@ public function up(): void
             $table->string('employee_name');
             $table->date('start_date');
             $table->date('end_date');
-            $table->unsignedSmallInteger('leave_type'); 
+            $table->enum('leave_type', ['Sick_Leave', 'Vacation_Leave', 'Annual_Leave', 'Maternity_Leave']);
             $table->text('reason')->nullable();
-            $table->unsignedSmallInteger('leave_status'); 
+            $table->enum('leave_status', ['Pending', 'Approved', 'Rejected']);
```
Comment (Kelv1nG) [inline src/database/migrations/2024_11_12_131427_create_leave_applications_table.php:17]: use values of enum directly here

$table->enum('column_name', array_column(MyEnum::cases(), 'value'));

**Pull request merged**